kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -217,18 +229,28 @@ public Transaction build(Engine engine) {
         snapshot,
         engineInfo,
         operation,
-        protocol,
-        metadata,
+        newProtocol.orElse(snapshotProtocol),
+        newMetadata.orElse(snapshotMetadata),
         setTxnOpt,
-        shouldUpdateMetadata,
-        shouldUpdateProtocol,
+        newMetadata.isPresent() /* shouldUpdateMetadata */,
+        newProtocol.isPresent() /* shouldUpdateProtocol */,
         maxRetries,
         table.getClock(),
         getDomainMetadatasToCommit(snapshot));
   }
 
-  /** Validate the given parameters for the transaction. */
-  private void validate(Engine engine, SnapshotImpl snapshot, boolean isNewTable) {
+  /**
+   * Validates the transaction as built given the parameters input by the user. This includes
+   *
+   * <ul>
+   *   <li>We can write to the current version of the table
```

Comment:
`We can write to the current version of the table` --> if the latest version of the table is `N`, we are not writing to version `N`. Could you clarify this a bit? i.e. I think you mean to express: `Ensures that the table, as defined by its latest protocol and metadata, is writable by Kernel.`